### PR TITLE
[FIX] l10n_in: fixes warning issue & allow the user to confirm the ve…

### PR DESCRIPTION
**Pull request by suth-odoo**
…ndor bill

Steps to Reproduce :
- install indian Accounting module
- click on invoice
- go to vendor bills
- create new

Issue:
- while creating new and confirming, it will throw a warning message,  as this warning required only for eInvoice only (while confirming the invoice) not for vendor Bills.

Cause:
- while generating warning message there is no specific condition like that it is not for vendors

Solution:
- if we gave condition that this warning message is only for out_invoice then the issue will be solved.

task-3657558

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jco-odoo** [comment]
@jva-odoo lgtm, let me know

File: addons/l10n_in/models/account_invoice.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def _post(self, soft=True):
         posted = super()._post(soft)
         gst_treatment_name_mapping = {k: v for k, v in
                              self._fields['l10n_in_gst_treatment']._description_selection(self.env)}
-        for move in posted.filtered(lambda m: m.country_code == 'IN'):
+        for move in posted.filtered(lambda m: m.country_code == 'IN' and m.move_type == 'out_invoice'):
```
Comment (JulienVR) [inline addons/l10n_in/models/account_invoice.py:87]: shouldn't we use the fiscal country code instead of the country code ?
Also, if those checks are only used for the EDI, they should preferably be put in `_check_move_configuration` I think, but anyway.
Why targeting master ? This line is also there in 15.0

Comment (suth-odoo) [reply addons/l10n_in/models/account_invoice.py]: Hello @JulienVR 
Thanks for the review
I have updated the changes
and also I retarget the branch to 15.0
Please have a look once if free
Thanks

Comment (JulienVR) [reply addons/l10n_in/models/account_invoice.py]: ping @smetl
Could I have your opinion on this ? Do you think we should use the country_code or the fiscal_country_code ?

Comment (jco-odoo) [reply addons/l10n_in/models/account_invoice.py]: Does not country_code already go to fiscal_country_code (which is on the company)

File: addons/l10n_in/models/account_invoice.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def _post(self, soft=True):
         posted = super()._post(soft)
         gst_treatment_name_mapping = {k: v for k, v in
                              self._fields['l10n_in_gst_treatment']._description_selection(self.env)}
-        for move in posted.filtered(lambda m: m.country_code == 'IN'):
+        for move in posted.filtered(lambda m: m.company_id.account_fiscal_country_id.code == 'IN' and m.move_type in ['out_invoice', 'out_refund']):
```
Comment (jco-odoo) [inline addons/l10n_in/models/account_invoice.py:104]:
```suggestion
        for move in posted.filtered(lambda m: m.country_code == 'IN' and m.move_type in ['out_invoice', 'out_refund']):
```
Unnecessary change. 

It is weird because there is code below that checks if the journal is purchase.

Comment (JulienVR) [reply addons/l10n_in/models/account_invoice.py]: It's better to use `m.is_sale_document()` instead of `m.move_type in ['out_invoice', 'out_refund']` :)

File: addons/l10n_in/models/account_invoice.py (hunk 1 above)
Comment (jco-odoo) [reply addons/l10n_in/models/account_invoice.py]: @JulienVR That field is related to the fiscal country code.

Comment (JulienVR) [reply addons/l10n_in/models/account_invoice.py]: you're right, good then, thanks ! :sweat_smile:

File: addons/l10n_in/models/account_invoice.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ def _post(self, soft=True):
         posted = super()._post(soft)
         gst_treatment_name_mapping = {k: v for k, v in
                              self._fields['l10n_in_gst_treatment']._description_selection(self.env)}
-        for move in posted.filtered(lambda m: m.country_code == 'IN'):
+        for move in posted.filtered(lambda m: m.company_id.account_fiscal_country_id.code == 'IN' and m.is_sale_document()):
```
Comment (JulienVR) [inline addons/l10n_in/models/account_invoice.py:104]: As pointed by @jco-odoo, `country_code` on the account.move is a related of `company_id.account_fiscal_country_id.code` (see the field definition).
So my first comment was wrong, you can just do:
```suggestion
        for move in posted.filtered(lambda m: m.country_code == 'IN' and m.is_sale_document()):
```
Sorry for the back and forths

**Comment by smetl** [comment]
@robodoo r+
